ballerina/utils.bal

```diff
@@ -15,21 +15,154 @@
+    Uuid u = check toRecord(uuid);
+
+    int mostSigBits = u.timeLow & 0xffffffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeMid & 0xffff;
+    mostSigBits <<= 16;
+    mostSigBits |= u.timeHiAndVersion & 0xffff;
+
+    return mostSigBits;
+}
+
+isolated function getLeastSignificantBits(string uuid) returns int|Error {
+    Uuid u = check toRecord(uuid);
+
+    int leastSigBits;
+    string clockSeq = regex:split(uuid, "-")[3];
+    int|error clockSeqInt = ints:fromHexString(clockSeq);
+    if clockSeqInt is int {
+        leastSigBits = clockSeqInt & 0xffff;
+    } else {
+        return error Error("Failed to get clock sequence value of the uuid");
+    }
+
+    leastSigBits <<= 48;
+    leastSigBits |= u.node & 0xffffffffffff;
+
+    return leastSigBits;
+}
+
+isolated function get64LeastSignificantBitsForVersion1() returns int {
+    int random63BitLong = nextLong() & 0x3FFFFFFFFFFFFFFF;
+    int variant3BitFlag = -0x8000000000000000;
+    return random63BitLong + variant3BitFlag;
+}
+
+isolated function get64MostSignificantBitsForVersion1() returns int {
+    // The gregorian time(1582-10-15 0:0:0) is a constant
+    int gregorianTimeInSeconds = 12219292800;
+    time:Utc duration = time:utcAddSeconds(time:utcNow(), <decimal>gregorianTimeInSeconds);
+    int timeForUuidIn100Nanos = duration[0] * 10000000 + <int>duration[1] * 100;
+    int least12SignificantBitOfTime = (timeForUuidIn100Nanos & 0x000000000000FFFF) >> 4;
+    int uuidVersion = 1 << 12;
+    return (timeForUuidIn100Nanos & 0x7FFFFFFFFFFF0000) + uuidVersion + least12SignificantBitOfTime;
+}
+
+isolated function formatUnsignedInt(int val, int shift, byte[] buf, int offset, int len) returns byte[] {
+    int result = val;
+    int charPos = offset + len - 1;
+    int radix = 1 << shift;
+    int mask = radix - 1;
+
+    while (charPos > offset - 1) {
+        lock {
+            buf[charPos] = digits[result & mask].toBytes()[0];
+        }
+        result >>>= shift;
+        charPos -= 1;
+    }
+    return buf;
+}
+
+isolated function getBytesFromSignificantBits(int msb, int lsb) returns byte[] {
+    byte[] result = [];
+
+    result[0] = <byte>((msb >> 56) & 0xff);
+    result[1] = <byte>((msb >> 48) & 0xff);
+    result[2] = <byte>((msb >> 40) & 0xff);
+    result[3] = <byte>((msb >> 32) & 0xff);
+    result[4] = <byte>((msb >> 24) & 0xff);
+    result[5] = <byte>((msb >> 16) & 0xff);
+    result[6] = <byte>((msb >> 8) & 0xff);
+    result[7] = <byte>((msb >> 0) & 0xff);
+
+    result[8] = <byte>((lsb >> 56) & 0xff);
+    result[9] = <byte>((lsb >> 48) & 0xff);
+    result[10] = <byte>((lsb >> 40) & 0xff);
+    result[11] = <byte>((lsb >> 32) & 0xff);
+    result[12] = <byte>((lsb >> 24) & 0xff);
+    result[13] = <byte>((lsb >> 16) & 0xff);
+    result[14] = <byte>((lsb >> 8) & 0xff);
+    result[15] = <byte>((lsb >> 0) & 0xff);
+
+    return result;
+}
+
+isolated function getUuidFromSignificantBits(int mostSigBits, int leastSigBits) returns string {
```

Comment:
'bitsToUuid' ?

Reply:
Fixed in [03747ed](https://github.com/ballerina-platform/module-ballerina-uuid/pull/376/commits/03747ed1608a8a8c02e488a630f335980ea719cc)